### PR TITLE
ham/eng 363 wallet age api check

### PR DESCRIPTION
**Pull request by ham-evans**
- wallet age api
- database wallet_age_check + mint checking
- fixing packs error

### CODE REVIEW

**Comment by ham-evans** [comment]
TODO: Add column to packs table in database, send to pages/pack/[packName].tsx, pass through to minter

File: lib/fetchers.ts [hunk 1]
```diff
@@ -83,3 +83,14 @@ export const verify = async (
...
+
+export const walletAgeChecker = async (account: string, chainId: number) => {
+  // const url = `/api/minter/check-claimed?account=${account}&tokenId=${tokenId?.toString()}`;
+  const url = new URL("minter/check-wallet-age", ikApiUrlBase);
+  url.searchParams.set("account", account);
+  url.searchParams.set("chainId", chainId.toString());
+
+  const response = await fetch(url);
+  if (response.ok) return await response.json();
+  else throw await response.text();
```
Comment (illepic) [inline lib/fetchers.ts]: No need for `else`, no need to await on `response.text()`

```ts
if (response.ok) return await response.json();

throw response.text();
```

File: pages/api/minter/check-wallet-age.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  const chainIdInt = parseInt(chainId, 10);
+
+  const rpcURL = chainRPCLookup[chainIdInt];
+
+  const provider = new ethers.providers.JsonRpcProvider(rpcURL);
+  const walletTxCount = await provider.getTransactionCount(account);
+  if (walletTxCount === 0) return res.json({ approved: false }); // ETH will blow up if 0
+
+  let approved;
```
Comment (illepic) [inline pages/api/minter/check-wallet-age.ts]: Delete this line (see below for more)

File: pages/api/minter/check-wallet-age.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    const etherscanProvider = new ethers.providers.EtherscanProvider(
+      undefined,
+      process.env.ETHERSCAN_API_KEY
+    );
+    const oldestTransaction = (await etherscanProvider.getHistory(account))[0]
+      .blockNumber;
+    const currentBlock = await provider.getBlockNumber();
+    const age = oldestTransaction ? currentBlock - oldestTransaction : 0;
+
+    approved = age > 5760 ? true : false;
```
Comment (illepic) [inline pages/api/minter/check-wallet-age.ts]: Instead: 
```ts
return res.json({ approved: age > 5760 });
```

File: pages/api/minter/check-wallet-age.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  } else {
+    approved = walletTxCount > 1;
+  }
```
Comment (illepic) [inline pages/api/minter/check-wallet-age.ts]: Now delete this.

File: pages/api/minter/check-wallet-age.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  res.json({ approved });
```
Comment (illepic) [inline pages/api/minter/check-wallet-age.ts]: Now change this to:
```ts
return res.json({ approved: walletTxCount > 1 });
```

**Review by illepic — DISMISSED** [review]

**Review by chair28980 — APPROVED** [review]

**Pull request merged**